pkg/cloudprovider/gcp/object_store.go

```diff
@@ -194,11 +218,37 @@ func (o *ObjectStore) DeleteObject(bucket, key string) error {
 	return errors.Wrapf(o.client.Bucket(bucket).Object(key).Delete(context.Background()), "error deleting object %s", key)
 }
 
+/*
+ * Use the iamSignBlob api call to sign the url if there is no credentials file to get the key from.
+ * https://cloud.google.com/iam/credentials/reference/rest/v1/projects.serviceAccounts/signBlob
+ */
+func (o *ObjectStore) SignBytes(bytes []byte) ([]byte, error) {
+	name := "projects/-/serviceAccounts/" + o.googleAccessID
+	resp, err := o.iamSvc.Projects.ServiceAccounts.SignBlob(name, &iamcredentials.SignBlobRequest{
+		Delegates: []string{name},
+		Payload:   string(bytes),
```

Comment:
looks like we need to base64-encode the bytes here -- `Payload: base64.StdEncoding.EncodeToString(bytes),` should do it